lib/inventory/metadata/metadata.go

```diff
@@ -313,12 +313,37 @@ func (c *fetchConfig) fetchCloudEnvironment(ctx context.Context) string {
 // the instance is running on AWS.
 // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
 func (c *fetchConfig) awsHTTPGetSuccess(ctx context.Context) bool {
-	url := "http://169.254.169.254/latest/meta-data/"
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	url := "http://169.254.169.254/latest/api/token"
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, url, nil)
+	if err != nil {
+		return false
+	}
```

Comment:
I would rather use the SDK for this
https://aws.github.io/aws-sdk-go-v2/docs/sdk-utilities/ec2-imds/

It has support for ipv4/ipv6 and a fallback to imds v1

Reply:
There are several other places where we hardcode the metadata service already. I have no issues with moving to the SDK where possible, but I'd rather handle that separately if possible. Would that be OK with you?

Also, fallback to IMDSv1 on AWS isn't needed. There are no cases where IMDSv1 is supported but IMDSv2 isn't. There are, however, cases where only IMDSv2 is supported (and IMDSv1 does not work).